src/test/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerActorTest.scala

```diff
@@ -294,20 +293,19 @@ class InstanceTrackerActorTest extends AkkaUnitTest with Eventually {
 
         And("internal state is updated")
         probe.send(f.taskTrackerActor, InstanceTrackerActor.List)
-        probe.expectMsg(InstanceTracker.InstancesBySpec.forInstances(staged, runningTwo))
+        probe.expectMsg(InstanceTracker.InstancesBySpec.empty)
       }
 
       "fails after failure during repository call to expunge" in {
         val f = new Fixture
         Given("an task instance tracker with initial state")
         val appId: PathId = PathId("/app")
-        val staged = TestInstanceBuilder.newBuilder(appId).addTaskStaged().getInstance()
-        val runningOne = TestInstanceBuilder.newBuilder(appId).addTaskRunning().getInstance()
-        val runningTwo = TestInstanceBuilder.newBuilder(appId).addTaskRunning().getInstance()
-        val appDataMap = InstanceTracker.InstancesBySpec.forInstances(staged, runningOne, runningTwo)
+        val running = TestInstanceBuilder.newBuilder(appId).addTaskRunning().getInstance()
+        val runningDecomissioned = running.copy(state = running.state.copy(goal = Goal.Decommissioned))
+        val appDataMap = InstanceTracker.InstancesBySpec.forInstances(runningDecomissioned)
         f.taskLoader.load() returns Future.successful(appDataMap)
 
-        When("a new staged task gets added")
```

Comment:
I don't quite understand this change. Why did we drop the staged instance? Why are we not covering the state of adding a staged instance?

Reply:
the staged instance is still there, just for this to test expunge, we had to change the goal to decomissioned. I can change this comment but I think that the fact that it is in staging is not that interesting as that it is in decommissioned